src/main/java/com/schedule/workout/workoutScheduler/services/RolesService.java

```diff
@@ -0,0 +1,88 @@
+package com.schedule.workout.workoutScheduler.services;
+
+import com.schedule.workout.workoutScheduler.controller.role.CreateRoleModel;
+import com.schedule.workout.workoutScheduler.controller.role.RoleModel;
+import com.schedule.workout.workoutScheduler.controller.role.UpdateRoleModel;
+import com.schedule.workout.workoutScheduler.database.IRolesRepository;
+import com.schedule.workout.workoutScheduler.database.model.RoleDB;
+import com.schedule.workout.workoutScheduler.exceptions.RoleAlreadyExists;
+import com.schedule.workout.workoutScheduler.exceptions.RoleNotFoundException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+@Service
+public class RolesService {
+
+    @Autowired
+    IRolesRepository rolesRepository;
+
+
+    public CreateRoleModel createRole(CreateRoleModel createRoleModel) {
+        List<RoleDB> roles = rolesRepository.findByName(createRoleModel.getName());
+        RoleDB roleDB = new RoleDB();
+        roleDB.setId(UUID.randomUUID().toString());
+        roles.forEach(role -> {
+            String existingRoleName = role.getName();
+            if (existingRoleName.equals(createRoleModel.getName())) {
+                throw new RoleAlreadyExists();
+            }
+
+        });
+        roleDB.setName(createRoleModel.getName());
+        roleDB.setDescription(createRoleModel.getDescription());
+        rolesRepository.save(roleDB);
+        return new CreateRoleModel(createRoleModel.getName(), createRoleModel.getDescription());
+    }
+
+    public UpdateRoleModel updateRole(String id, UpdateRoleModel updateRoleModel) {
+        if (rolesRepository.existsById(id)) {
+            RoleDB roleDBtoUpdate = rolesRepository.findById(id).get();
+
+            roleDBtoUpdate.setName(updateRoleModel.getName());
```

Comment:
If you have a role "Admin" in database, and in here updateRoleModel name is "Admin", it will be possible to have two roles with same name. fix that so role can only be saved or updated if its unique name